### DISCUSSION TITLE
build constraints error when trying to build or run helloworld

### DISCUSSION THREAD

**Discussion by BigHoss** [fyne-io/fyne#3628] [Q&A] [answered]
As i get the exact same error on two windows machines I am out of ideas how I could fix that. (even googling for 2 days didn't bring up anything useful)

i followed the [tutorial](https://developer.fyne.io/started/)

if i try to build our run my main.go or the demo i just get the following error

![image](https://user-images.githubusercontent.com/932436/216717764-b7fca861-1a72-49a5-89b5-c7e539676dad.png)

**Comment by Bluebugs** [comment, marked as answer]
What C compiler are you using?

**Reply by Jacalz** [reply]
It could also be useful to know which architecture and operating system it is.

**Reply by BigHoss** [reply]
Msys2,

Win11

My mistake was that i tried to use another terminal. I didn't think that i have to use the msys2 one. I am trying to figure that out. 

But for now. I can run it in that terminal.

**Reply by andydotxyz** [reply]
That's correct you need to use the terminal that is installed / configured in the C compiler install.
Configuring a developers preferred terminal/setup is too complicated for our documents as the variations are huge. We tried but found it was not possible to write something that worked widely enough.

**Reply by BigHoss** [reply]
I managed to get around having to have a separate terminal window open by using wsl2.

thanks for the help
